store/postgres/src/connection_pool.rs

```diff
@@ -1107,12 +1101,31 @@ impl PoolInner {
 
 embed_migrations!("./migrations");
 
+struct MigrationCount {
```

Comment:
IMO this struct could be a bit more clear , both`had_migrations` and `is_new` sort of require reading the code and usage to understand what is going, I wonder if an enum of NoChanges, NewShard, MigrationsApplied(Int) would make it a bit more clear. It's private and super critical so I will approve and leave it with you to decide if you think there is a benefit or not 🤓

Reply:
I agree that the naming isn't great. I like the idea of making this an `enum` - I'll do that next time I have to touch something in there.